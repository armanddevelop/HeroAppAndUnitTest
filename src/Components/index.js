export { HeroList } from "./Hero/HeroList";

export { Hero } from "./Hero/Hero";

export { Navbar } from "./UI/NavBar";

export { Login } from "./Login/Login";

export { SearchScreen } from "./Search/SearchScreen";

export { DcScreen } from "./DC/DcScreen";

export { MarvelScreen } from "./Marvel/MarvelScreen";
