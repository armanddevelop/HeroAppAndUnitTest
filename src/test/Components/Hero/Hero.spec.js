import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { Hero } from "../../../Components";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));
describe("Test in Hero component", () => {
  const container = mount(
    <MemoryRouter initialEntries={["/hero"]}>
      <Routes>
        <Route path="/hero" element={<Hero />} />
        <Route path="/" element={<h1>No Hero</h1>} />
      </Routes>
    </MemoryRouter>
  );
  it("Snapshot to Hero component  without hero params", () => {
    expect(container).toMatchSnapshot();
    expect(container.find("h1").text().trim()).toBe("No Hero");
  });
  it("Snapshot to Hero component with hero params", () => {
    const hero = "dc-batman";
    const container = mount(
      <MemoryRouter initialEntries={[`/hero/${hero}`]}>
        <Routes>
          <Route path="hero/:heroId" element={<Hero />} />
          <Route path="/" element={<h1>No Hero</h1>} />
        </Routes>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    expect(container.find("h3").text().trim()).toBe("Batman");
    expect(container.find("img").prop("src")).toBe(
      `/assets/heroes/${hero}.jpg`
    );
  });
  it("hero test handleReturn", () => {
    const hero = "dc-batman";
    const container = mount(
      <MemoryRouter initialEntries={[`/hero/${hero}`]}>
        <Routes>
          <Route path="hero/:heroId" element={<Hero />} />
          <Route path="/" element={<h1>No Hero</h1>} />
        </Routes>
      </MemoryRouter>
    );

    container.find("button").prop("onClick")();

    expect(mockNavigate).toHaveBeenCalledWith("/dc", { replace: true });
  });
});
