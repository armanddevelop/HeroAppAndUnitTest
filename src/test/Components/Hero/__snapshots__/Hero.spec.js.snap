// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Hero component Snapshot to Hero component  without hero params 1`] = `
<h1>
  No Hero
</h1>
`;

exports[`Test in Hero component Snapshot to Hero component with hero params 1`] = `
<div
  className="row-mt-5"
>
  <h1>
    Bruce Wayne
  </h1>
  <div
    className="col-4"
  >
    <img
      alt="Batman"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/heroes/dc-batman.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
      Batman
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <b>
          Aleter ego:
        </b>
         
        Bruce Wayne
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Bio:
        </b>
        <p>
          Batman es un personaje de cómic creado por los estadounidenses Bob Kane y Bill Finger, ​ y propiedad de DC Comics. Apareció por primera vez en la historia titulada «El caso del sindicato químico» de la revista Detective Comics N.º 27, lanzada por la editorial National Publications el 30 de marzo de 1939
        </p>
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher:
        </b>
        <p>
          DC Comics
        </p>
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First Appearence:
        </b>
        <p>
          Detective Comics #27
        </p>
      </li>
    </ul>
    <h5
      className="mt-3"
    >
      Characters
    </h5>
    <p>
      Bruce Wayne
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
